packages/scaffold/src/modal/w3m-modal/index.ts

```diff
@@ -25,16 +27,56 @@ export class W3mModal extends LitElement {
   // -- State & Properties -------------------------------- //
   @state() private open = ModalController.state.open
 
+  @state() private caipAddress = AccountController.state.address
+
+  @state() private isSiweEnabled = SIWEController.state.isSiweEnabled
+
   public constructor() {
     super()
     this.initializeTheming()
     ApiController.prefetch()
     this.unsubscribe.push(
-      ModalController.subscribeKey('open', val => (val ? this.onOpen() : this.onClose()))
+      ModalController.subscribeKey('open', val => (val ? this.onOpen() : this.onClose())),
+      SIWEController.subscribeKey('isSiweEnabled', isEnabled => {
+        this.isSiweEnabled = isEnabled
+      }),
+      AccountController.subscribe(async newState => {
+        const { isConnected, caipAddress: newCaipAddress } = newState
```

Comment:
nit: pack this logic into a private method and use it as callback for this subscribe, to keep constructor "lean".